### PR TITLE
Move licenses to LICENSES/, use SPDX-License-Identifier, mention all licenses in galaxy.yml

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Implements https://github.com/ansible-community/community-topics/issues/112#issuecomment-1166348220.

Closes #393.

##### ISSUE TYPE
- Docs Pull Request
- Feature Pull Request

##### COMPONENT NAME
collection

### CODE REVIEW

File: .azure-pipelines/README.md
Comment (gotmax23) [inline .azure-pipelines/README.md:3]: I would use HTML style comments, instead. e.g.

``` markdown
<!--
GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
SPDX-License-Identifier: GPL-3.0-or-later
-->
```

Take a look at https://github.com/ansible-collections/community.docker/blob/b60d7d61b0a5583db185b3b5a4487846fcd94c56/.azure-pipelines/README.md to see how this currently looks

Comment (gotmax23) [reply .azure-pipelines/README.md:3]: I'll look at this more later. That was just the first thing that I noticed.

Comment (felixfontein) [reply .azure-pipelines/README.md]: I was using https://stackoverflow.com/a/20885980 as a reference. Maybe it needs another newline between the two comments, I'll try that. (one minute later) Hmm, that doesn't work either (https://github.com/ansible-collections/community.docker/blob/f1fed46fb528934f6ee510354d937571b2d45257/README.md).

Ok, a bit more experimenting showed that the brakets around the comment text are important. But once I add these, reuse cannot parse `[comment]: # (SPDX-License-Identifier: GPL-3.0-or-later)` anymore: `reuse._util - ERROR - Could not parse 'GPL-3.0-or-later)'`. What a mess. I've switched to HTML comments now, I simply hope that this will also work with Galaxy's README Markdown renderer...

Comment (gotmax23) [reply .azure-pipelines/README.md]: >  I've switched to HTML comments now, I simply hope that this will also work with Galaxy's README Markdown renderer...

[Have your worries assuaged](https://galaxy.ansible.com/gotmax23/aur_test) :)

File: tests/sanity/ignore-2.10.txt
Comment (gotmax23) [inline tests/sanity/ignore-2.10.txt]: REUSE also supports creating a standalone `/path/to/file`.license file with just `SPDX-License-Identifier: GPL-3.0-or-later` (no comment needed). See `reuse addheader --explict` and https://reuse.software/spec/#comment-headers.

File: .azure-pipelines/README.md
Comment (felixfontein) [reply .azure-pipelines/README.md]: Thanks for testing this!

File: tests/sanity/ignore-2.10.txt
Comment (felixfontein) [reply tests/sanity/ignore-2.10.txt]: I've read about it, but that looks a lot more ugly to me :-)

File: plugins/module_utils/version.py
Comment (gotmax23) [inline plugins/module_utils/version.py:3]: `reuse lint` does not like the colon after `# Copyright`. I would remove this to please it and to make this consistent with the other headers, but I'll leave it up to you :)

Comment (gotmax23) [inline plugins/module_utils/version.py:3]: `reuse lint` does not like the colon after `# Copyright`. I would remove this to please it and to make this consistent with the other headers, but I'll leave it up to you :)

File: plugins/modules/docker_plugin.py
Comment (gotmax23) [inline plugins/modules/docker_plugin.py:5]: Same here.

**Comment by gotmax23** [comment]
Output of `reuse lint` (with the colons from the copyrights removed):

---

reuse._util - ERROR - Could not parse '')'
reuse.project - ERROR - 'tests/sanity/extra/licenses.py' holds an SPDX expression that cannot be parsed, skipping the file
# MISSING COPYRIGHT AND LICENSING INFORMATION

The following files have no copyright and licensing information:
* CHANGELOG.rst
* changelogs/changelog.yaml
* changelogs/fragments/430-licenses.yml
* tests/sanity/extra/extra-docs.json
* tests/sanity/extra/licenses.json
* tests/sanity/extra/licenses.py -- **False negative, thrown off by the other SPDX-License-Identifer text**
* tests/sanity/extra/no-unwanted-files.json
* tests/sanity/ignore-2.10.txt
* tests/sanity/ignore-2.11.txt
* tests/sanity/ignore-2.12.txt
* tests/sanity/ignore-2.13.txt
* tests/sanity/ignore-2.14.txt
* tests/sanity/ignore-2.9.txt


# SUMMARY

* Bad licenses:
* Deprecated licenses:
* Licenses without file extension:
* Missing licenses:
* Unused licenses:
* Used licenses: Apache-2.0, GPL-3.0-or-later
* Read errors: 0
* Files with copyright information: 432 / 445
* Files with license information: 432 / 445

Unfortunately, your project is not compliant with version 3.0 of the REUSE Specification :-(

---

REUSE has other ways that you can add licenses to these files, but I'm fine with this implementation. All of the actual code has explicit license identifiers and copyrights.

**Review by gotmax23 — APPROVED** [review]
Minus some minor quibbles, which you can choose how to address, this looks good to me. I didn't inspect the diff for every single file, but I ran this through your linting tool and `reuse lint` and checked many of the diffs.

File: plugins/module_utils/version.py
Comment (felixfontein) [reply plugins/module_utils/version.py]: Fixed in b6721b6bb7d1e4136cef1218ef3d033899057aa9.

File: plugins/modules/docker_plugin.py
Comment (felixfontein) [reply plugins/modules/docker_plugin.py]: Fixed in b6721b6bb7d1e4136cef1218ef3d033899057aa9.

**Review by gotmax23 — COMMENTED** [review]
I looked over this again and found a couple other things to nitpick on, but over all I'm very happy with this! I'm happy to see my suggestion getting implemented :blush:. Thanks for all your work on this and for writing the tools to make the process easier.

File: tests/sanity/extra/licenses.py
Comment (felixfontein) [reply tests/sanity/extra/licenses.py:7]: No, but `ansible-test sanity` will complain if this isn't present in a Python file.

File: README.md
Comment (felixfontein) [reply README.md:122]: The problem is that the relative links will be broken in the main place where these links are shown: on the collection's Galaxy page. So the link should better go to https://github.com/ansible-collections/community.docker/blob/main/LICENSES/Apache-2.0.txt (which works only once this is merged). Can you adjust your suggestion?

Comment (felixfontein) [reply README.md:122]: Also `Each file in the collection that is not automatically generated has a machine readable [...]` is not correct. There are some manually created files that do not have such a comment, such as `tests/sanity/ignore-2.*.txt` and `tests/sanity/extra/*.json` and `tests/integration/targets/luks_device/files/keyfile*`. The license files themselves also do not have one. How about

> Most files in the collection that are not automatically generated have a machine readable `SDPX-License-Identifier:` comment denoting its license(s).

instead?

File: tests/sanity/extra/licenses.py
Comment (felixfontein) [reply tests/sanity/extra/licenses.py]: Improved in a83a7d5b72f6241fd9f65dd9b5510beb7d6b21bf.

File: tests/sanity/extra/licenses.py
Comment (felixfontein) [reply tests/sanity/extra/licenses.py:77]: Improved in a83a7d5b72f6241fd9f65dd9b5510beb7d6b21bf.

File: README.md
Comment (gotmax23) [reply README.md:122]: I've edited the suggestion

File: README.md
Comment (felixfontein) [inline README.md:120]:
```suggestion
See [LICENSES/GPL-3.0-or-later.txt](https://github.com/ansible-collections/community.docker/blob/main/COPYING) for the full text.
```

File: README.md
Comment (gotmax23) [inline README.md:120]: /me wishes Github would properly handle symlinks instead of creating useless other features and committing license violations en masse (Copilot), but this is probably the best solution.

Comment (felixfontein) [reply README.md:120]: :+1:

**Comment by felixfontein** [comment]
@gotmax23 thanks a lot for initiating and reviewing this!

**Pull request merged**
